cmake/SetupBlas.cmake

```diff
@@ -16,3 +16,20 @@ set_property(
   GLOBAL APPEND PROPERTY SPECTRE_THIRD_PARTY_LIBS
   Blas
   )
+
+# Check if we have found OpenBLAS and can disable its multithreading, since it
+# conflicts with Charm++ parallelism. Details:
+# https://github.com/xianyi/OpenBLAS/wiki/Faq#multi-threaded
+try_compile(
```

Comment:
I specifically requested you don't read a file from disk...

Reply:
You said "writing a file to disk". I'm not sure this can be done without reading a file from disk (see @fmahebert's and my comment above).

Reply:
Yes, it can be done...

Reply:
See my comment "We do this for warning flags"

Reply:
This is the code in `cmake/CheckCxxLinkerFlag.cmake`, if that's what you mean...

```cpp
  set(SOURCE_TO_TRY "int main() { return 0; }")
  set(FILE_TO_WRITE
    "${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/CMakeTmp/src.cxx")
  file(WRITE "${FILE_TO_WRITE}" "${SOURCE_TO_TRY}\n")
  set(LINK_COMMAND_TO_TRY
    "${CHARM_COMPILER} ${FLAG_TO_CHECK} <CMAKE_CXX_LINK_FLAGS> <LINK_FLAGS> <OBJECTS> -o <TARGET> <LINK_LIBRARIES>")

  if(NOT CMAKE_REQUIRED_QUIET)
    message(STATUS "Performing Test ${FLAG_TO_CHECK}")
  endif()

  try_compile(${FLAG_WORKS}
    ${CMAKE_BINARY_DIR}
    ${FILE_TO_WRITE}
    CMAKE_FLAGS -DCMAKE_CXX_LINK_EXECUTABLE:STRING=${LINK_COMMAND_TO_TRY}
    OUTPUT_VARIABLE OUTPUT)
```

Reply:
You're looking at **linker** flags, not compiler flags... `AddCxxFlag.cmake`

Reply:
I see, that's using `execute_process` instead of `try_compile`. Seems rather painful to do, linking with Blas and everything, when `try_compile` works fine and doesn't notably increase cmake configuration time..

Reply:
I'll have to look at this again another time, it's late

Reply:
Please use the approach from `AddCxxFlag`. There's a reason we do that: using `try_compile` is very expensive and using it many times really slows down CMake. This is why we do the flags the way they are done. The argument "Using it only once here..." can be applied repeatedly and then CMake takes forever to run. Also keep in mind that, especially on HPC systems, the filesystem can be very slow and while you might not notice a slowdown on your machine with an SSD, that will almost certainly not be true on some network file systems.